docs/errors/E0373.md

```diff
@@ -0,0 +1,21 @@
+# E0373: mistyped strict inequality operator
+
+While `x! == y` is legal TypeScript for a non-null assertion, it may be a typo for a strict inequality, as in `x !== y`:
+
+```javascript
+let x = 17;
+let y = 38;
+if (x! == y) {
```

Comment:
Must fix: CI caught an issue with this code: This example reports E0373 (as intended), but also reports E0261 (TypeScript non-null assertions are not allowed in JavaScript).

Please avoid reporting E0261 (diag_typescript_non_null_assertion_not_allowed_in_javascript) if E0373 (diag_mistyped_strict_inequality_operator) is reported.

Reply:
Should I move my implementation of the check before the JavaScript vs TypeScript check?
Or is there a way to change the JavaScript in the documentation into typescript as to not trigger the error?